packages/discovery-react-components/src/components/DocumentPreview/components/PdfViewer/PdfViewer.tsx

```diff
@@ -19,7 +19,8 @@ type Props = PdfDisplayProps & {
   className?: string;
 
   /**
-   * PDF file data as base64-encoded string
+   * PDF file data as a "binary" string (array buffer)
+   * TODO Update to take `PDFSource` type (from pdfjs-dist) instead? Would allow binary data as well as URL.
```

Comment:
The function `getDocument()` from `pdfjs-dist` can take much more than just an array buffer. In particular, it can also take a URL, which would be a nice feature for the `DocumentProvider` interface. We should update this and associated files with the expanded type.

For reference, here's the [type](https://github.com/mozilla/pdfjs-dist/blob/bd2a6cd87f9e0c3a9b3197d5d27acfb99886629b/types/src/display/api.d.ts#L618-L620) as defined in the latest `pdfjs-dist` version:

```
 * @param {string|URL|TypedArray|PDFDataRangeTransport|DocumentInitParameters}
 *   src - Can be a URL where a PDF file is located, a typed array (Uint8Array)
 *         already populated with data, or a parameter object.
```

The version we're using (v2.2.228) instead defines an object named `PDFSource` with `url` and `data` props. But if we're going to expand this type in our files, we should follow the lead of the latest `pdfjs-dist` code.

Reply:
This is very good thing that we should consider. To align the latest `pdfjs-dist`, I'd add a new `src` prop and deprecate the `file` like below. However it wouldn't be small. So I think it'd good to leave this TODO for now and handle in a separated PR.

```
  /**
   * PDF data that PDFJS's getDocument can consume
   */
  src?: PDFSource; // Parameters<typeof PDFJSLib.getDocument>[0]
  /**
   * @deprecated PDF file data as a "binary" string (array buffer)
   * use `src={{data: <file prop value>}}` instead
   */
  file?: string;
```